### PR TITLE
Sonobuoy status

### LINKED ISSUE #209: Implement status subcommand

### PR DESCRIPTION
**Pull request by liztio**
Example status:
```
+--------------+-----------------------------+----------+
|    PLUGIN    |            NODE             |  STATUS  |
+--------------+-----------------------------+----------+
| e2e          |                             | running  |
| systemd_logs | ip-10-0-18-159.ec2.internal | complete |
| systemd_logs | ip-10-0-22-130.ec2.internal | complete |
| systemd_logs | ip-10-0-22-8.ec2.internal   | complete |
+--------------+-----------------------------+----------+
|                                              RUNNING  |
+--------------+-----------------------------+----------+
```

closes #209

### CODE REVIEW

**Comment by liztio** [comment]
I think adding the tablewriter greatly enhances usability for users, but we could do it more messily if we don't want the extra dependency.

File: cmd/sonobuoy/app/status.go [hunk 1]
```diff
@@ -19,28 +19,61 @@ package app
...
 	"os"
 
-	ops "github.com/heptio/sonobuoy/cmd/sonobuoy/app/operations"
-	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/olekukonko/tablewriter"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/heptio/sonobuoy/cmd/sonobuoy/app/args"
```
Comment (chuckha) [inline cmd/sonobuoy/app/status.go]: This shouldn't be pulling in args, see https://github.com/heptio/sonobuoy/pull/264

File: pkg/plugin/aggregation/update.go [hunk 2]
```diff
@@ -13,7 +13,10 @@ import (
 	"github.com/heptio/sonobuoy/pkg/plugin"
 )
 
-const annotationName = "sonobuoy.hept.io/status"
+const (
+	StatusAnnotationName = "sonobuoy.hept.io/status"
+	StatusPodName        = "sonobuoy"
+)
 
 type key struct {
```
Comment (chuckha) [inline pkg/plugin/aggregation/update.go]: Please document this composite key. Maybe something about how a plugin is uniquely defined by the node it runs on and its name (which isn't actually true as we would also need the sonobuoy id to uniquely define a plugin).

File: pkg/plugin/aggregation/update.go [hunk 3]
```diff
@@ -22,21 +25,20 @@ type key struct {
 // Updater manages setting the Aggregator annotation with the current status
 type Updater struct {
```
Comment (chuckha) [inline pkg/plugin/aggregation/update.go]: Does anything here need to be exported?

File: cmd/sonobuoy/app/status.go (hunk 1 above)
Comment (liztio) [reply cmd/sonobuoy/app/status.go]: I thought `cmd` was the right place for args and `operations` was the wrong place for args.

**Comment by liztio** [comment]
We don't support multiple sonobuoy runs in the same namespace, aiui. Parallel sonobuoy runs would be in different namespaces, and you pass in a `--namespace` flag for that reason.

File: cmd/sonobuoy/app/status.go (hunk 1 above)
Comment (chuckha) [reply cmd/sonobuoy/app/status.go]: yeah, my mistake.

**Comment by chuckha** [comment]
ah ha ok, knew i must have been missing something

File: pkg/plugin/aggregation/update.go (hunk 3 above)
Comment (liztio) [reply pkg/plugin/aggregation/update.go]: probably not!

**Comment by chuckha** [comment]
could we use https://golang.org/pkg/text/tabwriter/ instead of the tablewriter? would that still greatly enhance usability while not pulling in an external dep?

**Pull request merged**